### DISCUSSION TITLE
Dockerfile best practice

### DISCUSSION THREAD

**Discussion by gvanto** [pimcore/pimcore#17244] [Q&A / Support] [answered]
I'm looking to create a new Dockerfile for our pimcore setup, with the intention of running on AWS ECS.

I was wondering what is the best/most common practice for doing this?
- extend the official pimcore docker image (I have some customizations that need to run in the dockerfile so would need to FROM from it - but I am a [bit confused about a few things](https://stackoverflow.com/questions/78682389/dockerhub-pimcore-image))?

-create the image from scratch?

Pros/cons of the above 2 methods? To be honest right now leaning towards the custom image as I don't have any idea what is going on in the [layers of the official pimcore dockerhub image](https://registry.hub.docker.com/layers/pimcore/pimcore/php8.1-debug-latest/images/sha256-c2bba6e4dcc40ef8554778d139f986fe1cef2393410fb79e9ea7d9da6916a04d?context=explore)

Any help/advice much appreciated

**Comment by fcaps** [comment, marked as answer]
@gvanto i just used the provided image and locked the version `FROM pimcore/pimcore:php8.3-v3 `, if you have zero trust, then you can also lock to a hash . If you do not trust pimcore to provide proper and secure images, create your own by forking [that repo ](https://github.com/pimcore/docker/blob/3.x/Dockerfile) and build/push to your own registry. The images are kinda bloated, but that's the price you pay if you do not create your own image. 
Another consideration are updates and security fixes, do you have time to make those?

**Reply by gvanto** [reply]
Thanks @fcaps appreciate it.
